go/pserver/optimizer.go

```diff
@@ -1,42 +1,42 @@
 package pserver
 
 /*
-#include "optimizer.h"
+// TODO(zhihong): move compile flags to cmake go_library
+#cgo pkg-config: protobuf
+#cgo CFLAGS: -I ../../
+#cgo LDFLAGS: ../../build/paddle/optimizer/libpaddle_optimizer.a ../../build/proto/libpaddle_proto.a ../../third_party/install/glog/lib/libglog.a ../../third_party/install/gtest/lib/libgtest.a ../../third_party/install/gflags/lib/libgflags.a ../../third_party/install/openblas/lib/libopenblas.a -I/usr/local/lib/ -lprotobuf
+#include "paddle/optimizer/optimizer.h"
 */
 import "C"
 import (
 	"fmt"
 	"unsafe"
 )
 
-type optimizerType int
-
-const (
-	sgd optimizerType = iota
-)
-
 var nullPtr = unsafe.Pointer(uintptr(0))
 
 type optimizer struct {
 	opt *C.struct_paddle_optimizer
 }
 
-func newOptimizer(t optimizerType, learning_rate float64) *optimizer {
+func newOptimizer(paramWithConfigs ParameterWithConfig) *optimizer {
 	o := &optimizer{}
-	o.opt = C.paddle_create_SGD_optimizer(C.double(learning_rate))
+	p := paramWithConfigs.Param
+	c := paramWithConfigs.Config
+	o.opt = C.paddle_create_optimizer(C.uchar(c), C.int(len(c)), unsafe.Pointer(p.Content), c.int(p.Length), nullPtr, 0)
```

Comment:
I see. fix the copy done.
>Another question is what is the benefit of changing the type of p.Content from []byte to *byte:
there is no benefit,  c style language prefers pointer type.  fix it